src/zc/buildout/easy_install.py

```diff
@@ -169,6 +169,60 @@ def _execute_permission():
 
 _easy_install_cmd = 'from setuptools.command.easy_install import main; main()'
 
+def _get_source(filename):
+    if not os.path.isfile(filename):
+        return None
+    with open(filename, 'rb') as f:
+        return f.read()
+
+def get_namespace_package_paths(dist):
+    """
+    Generator of the expected pathname of each __init__.py file of the
+    namespaces of a distribution.
+    """
+    base = [dist.location]
+    init = ['__init__.py']
+    for namespace in dist.get_metadata_lines('namespace_packages.txt'):
+        yield os.path.join(*(base + namespace.split('.') + init))
+
+def get_namespace_package_sources(dist):
+    for path in get_namespace_package_paths(dist):
+        yield _get_source(path)
+
+def dist_needs_pkg_resources(dist):
+    """
+    A distribution needs setuptools/pkg_resources added as requirement if:
+
+        * It has namespace packages declared with:
+        - `pkg_resources.declare_namespace()`
+        * Those namespace packages don't fall back to `pkgutil`
+        * It doesn't have `setuptools/pkg_resources` as requirement already
+    """
```

Comment:
> Oh gawd, I'm regretting ever providing this crutch. :)

Wanna get rid of it? It's fine by me. I actually thought of ripping it out myself. Ended up making it more complex instead so it wouldn't be so eager to add setuptools to everything now that it's not needed.